connector/jobrunner/runner.py

```diff
@@ -131,6 +135,21 @@
 _logger = logging.getLogger(__name__)
 
 
+# Unfortunately, it is not possible to extend the Odoo
+# server command line arguments, so we resort to environment variables
+# to configure the runner (channels mostly).
+#
+# On the other hand, the odoo configuration file can be extended at will,
+# so we check it in addition to the environment variables.
+
+
+def _channels():
+    return (
+        os.environ.get('ODOO_CONNECTOR_CHANNELS') or
+        config.misc.get("options-connector", {}).get("channels")
```

Comment:
@sbidoul, you're right that the config file is more visible and more specific than env variables.

On the other hand, env variables are easier to change between invocations than config file values, as you can do it directly on the command line like:

    ODOO_CONNECTOR_CHANNELS=subchannel:5 bin/start_odoo

Also, some deployment systems and PaaS services make it easy to use the same setup (i.e. the same set of code and configuration) in different roles by changing environment variables.

Then again, if someone wants to use it like this, I guess they could just leave the config file empty...

I'd just like to point out that some systems use environment variables to override settings in the configuration file. One notable example is [git](https://git-scm.com/docs/git-config), where things like `user.name` and `user.email` from the configuration can conveniently be changed for the scope of a single commit by setting `GIT_AUTHOR_NAME` and `GIT_AUTHOR_EMAIL`.

In any case, @sbidoul, @pedrobaeza, I'm not dead-set against changing the priority. I'd just like to ask you to consider the above points one more time and give me a final opinion.

If you decide that environment should have priority after all, I'll add one final change to the code above to specify that, if the env var is set, even if it's empty, then it will cancel the channel configuration from the file, so that it's possible to disable the runner with an invocation like.

    ODOO_CONNECTOR_CHANNELS="" bin/start_odoo

Otherwise I'll swap things around.